### DISCUSSION TITLE
Select which folders to hash

### DISCUSSION THREAD

**Discussion by ghost** [ansani/Shareaza#82] [Ideas]
I have tried many p2p file-sharing clients but the only one in which I've been able to share all my files is Nicotine+. To be able to use a single program it would be nice if I could share my files without hashing on the SoulSeek network or other protocols that don't require hashing and select some checkboxes of some folders or subfolders to hash and share through the other networks.

**Comment by ansani** [comment]
A multi-network file-sharing client like Shareaza aims to download the same file (chunks) from different networks. To identify the file (or the chunk to request/download) in different protocols, the client needs a hash. The SoulSeek (and other related hashless protocols, you should read, for example, the thread about the implementation of IRC DCC) doesn't support the same interoperability; the only check we can do is about the name/size (not a check...). The implementation of protocols that cannot guarantee a strict check for interoperability has lower priority. Of course, I will create an issue for that!

**Reply by ghost** [reply]
This is a doubt I've had about hash interoperability since I learned that BitComet claims to improve torrent availability by using the ed2k network.

When you download a torrent you already know from the torrent file the hash of each of the chunks, but as far as I know you can't get the same hash for another network unless you have the chunk downloaded. So how can it download chunks from ed2k that uses an md4 hash algorithm having the hash for a md5 algorithm? It would need a database to search for the same file by the md5 hash and get the md4 hash of all the chunks, similar to the torrent for BitTorrent. But there isn't anything like this.

And similarly, how does Shareaza find chunks for a file from different networks without knowing the hash of the chunks for each network?
